Packs/DeepInstinct/Integrations/DeepInstinct/DeepInstinct.py

```diff
@@ -0,0 +1,324 @@
+    """
+    Remove hash from whitelist
+    """
+    policy_id = demisto.args().get('policy_id')
+    file_hash = demisto.args().get('file_hash')
+    http_request('DELETE', '/policies/%s/whitelist/hashes/%s' % (str(policy_id), file_hash))
+    demisto.results('ok')
+
+
+def add_devices_to_group():
+    """
+    Add devices to specific group
+    """
+    group_id = demisto.args().get('group_id')
+    device_ids_input = demisto.args().get('device_ids')
+    device_ids = [int(num) for num in device_ids_input.split(",")]
+    http_request('POST', '/groups/%s/add-devices' % str(group_id), json={"devices": device_ids})
+    demisto.results('ok')
+
+
+def remove_devices_from_group():
+    """
+    Remove devices from group
+    """
+    group_id = demisto.args().get('group_id')
+    device_ids_input = demisto.args().get('device_ids')
+    device_ids = [int(num) for num in device_ids_input.split(",")]
+
+    http_request('POST', '/groups/%s/remove-devices' % str(group_id), json={"devices": device_ids})
+    demisto.results('ok')
+
+
+def delete_files_remotely():
+    """
+    Delete given file ids remotely
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/devices/actions/delete-remote-files', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def terminate_remote_processes():
+    """
+    Terminate remove processes by given event ids
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/devices/actions/terminate-remote-process', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def close_events():
+    """
+    Close events by event ids
+    """
+    event_ids_input = demisto.args().get('event_ids')
+    event_ids = [int(num) for num in event_ids_input.split(",")]
+    http_request('POST', '/events/actions/close', json={"ids": event_ids})
+    demisto.results('ok')
+
+
+def fetch_incidents():
+    incidents = []
+    last_id = demisto.params().get('first_fetch_id')
+
+    last_run = demisto.getLastRun()
+    if last_run and last_run.get('last_id') is not None:
+        last_id = last_run.get('last_id')
+
+    events = http_request('GET', '/events/?after_id=' + str(last_id))
+    while events and events['events']:
+        for event in events['events']:
+            incident = {
+                'name': "DeepInstinct_" + str(event['id']),  # name is required field, must be set
+                'occurred': event['insertion_timestamp'],
+                'rawJSON': json.dumps(event)
+            }
+            incidents.append(incident)
+
+        demisto.setLastRun({'last_id': events['last_id']})
+        events = http_request('GET', '/events/?after_id=' + str(events['last_id']))
+
+    demisto.incidents(incidents)
+
+
+def test_module():
+    http_request('GET', "/health_check")
+    demisto.results("ok")
```

Comment:
As i talked to Eldad, we need a clear msg of what happened, does it come back from the API?
+
You are not covering a situation in which the URL is not the right one.
This is a pain-point for a lot of customers so please address this.

Reply:
1. you will see the message from http_request function
2. What url, base url? You will get authorization error if it's not  right. How can I know it's not right otherwise?

Reply:
I just want a clear error that if you have authorization error than the user will see something like:
`Authentication parameters are invalid, please check your URL address and your API toke`